pkg/kubectl/rolling_updater.go

```diff
@@ -308,14 +308,14 @@ func (r *RollingUpdater) scaleDown(newRc, oldRc *api.ReplicationController, desi
 		return oldRc, nil
 	}
 	// Block until there are any pods ready.
-	_, newAvailable, err := r.waitForReadyPods(config.Interval, config.Timeout, oldRc, newRc)
+	oldAvailable, newAvailable, err := r.waitForReadyPods(config.Interval, config.Timeout, oldRc, newRc)
```

Comment:
Does the replication controller select unready pods first?  I'm trying to run through my scenario here to see if this is what I expect:

1. deployment N with 2 pods, both unready
2. deployment N+1 with 1 pod, ready
3. controller was wedged because `oldRc.Spec.Replicas` was 2, not 0

With this change:

3. controller will get back 0 old available, 1 new available
4. min available will be 1
5. decrement will be zero?
6. controller will still be wedged

This may be brian's issue https://github.com/kubernetes/kubernetes/issues/16737#issuecomment-175242784 - basically if `oldAvailable < minAvailable` then we are ok to scale down by the difference between oldAvailable and `oldRc.Spec.Replicas` (although maybe we want to do it in successive steps, where we always do at least 1).